search/mongo_search_test.go

```diff
@@ -126,12 +133,19 @@ func (m *MongoSearchSuite) TestConditionCodeQueryObjectByCode(c *C) {
 }
 
 func (m *MongoSearchSuite) TestConditionCodeQueryByCode(c *C) {
-	var conditions []*models.Condition
 	q := Query{"Condition", "code=123641001"}
-	mq := m.MongoSearcher.CreateQuery(q)
-	err := mq.All(&conditions)
+
+	results, _, err := m.MongoSearcher.Search(q)
 	util.CheckErr(err)
-	c.Assert(conditions, HasLen, 2)
+	resultsVal := reflect.ValueOf(results).Elem()
+	c.Assert(resultsVal.Len(), Equals, 2)
+
+	// convert search results to conditions
+	conditions := make([]models.Condition, resultsVal.Len())
+	for i := 0; i < resultsVal.Len(); i++ {
+		conditions[i] = resultsVal.Index(i).Interface().(models.Condition)
+	}
+
```

Comment:
Not a big deal, but we've seen this pattern (above) already and we'll see it several more times below.  So... it probably could be made into some sort of helper function, like:
```
func (m *MongoSearchSuite) assertConditionResult(results interface{}, expectedSize int) []models.Condition
```